File: app/src/main/java/com/mxt/anitrend/model/entity/container/body/Container.java
package com.mxt.anitrend.model.entity.container.body;

import com.mxt.anitrend.model.entity.container.attribute.PageInfo;

abstract class Container {

    private PageInfo pageInfo;

    public PageInfo getPageInfo() {
        return pageInfo;
    }

    public boolean hasPageInfo() {
        return pageInfo != null;
    }

    public abstract boolean isEmpty();
}

File: app/src/main/java/com/mxt/anitrend/model/entity/container/body/Container.kt
package com.mxt.anitrend.model.entity.container.body

import com.mxt.anitrend.model.entity.container.attribute.PageInfo

abstract class Container {

    lateinit var pageInfo: PageInfo

    fun hasPageInfo(): Boolean = ::pageInfo.isInitialized

    abstract val isEmpty: Boolean
}
